query-creator/create-query/tests/unit/test-handler.js

```diff
@@ -115,280 +139,442 @@ describe('Denormalize resource', function () {
+          'Correct answer' AS student_id,
+          null AS user_id,
+          null AS student_name,
+          null AS username,
+          null AS school,
+          null AS class,
+          null AS class_id,
+          null AS permission_forms,
+          null AS teacher_user_ids,
+          null AS teacher_names,
+          null AS teacher_districts,
+          null AS teacher_states,
+          null AS teacher_emails,
+          null AS res_1_name,
+          null AS res_1_learner_id,
+          null AS res_1_remote_endpoint,
+          null AS res_1_resource_url,
+          null AS res_1_last_run,
+          null AS res_1_total_num_questions,
+          null AS res_1_total_num_answers,
+          null AS res_1_total_percent_complete,
+          null AS res_1_num_required_questions,
+          null AS res_1_num_required_answers,
+          null AS res_2_name,
+          null AS res_2_learner_id,
+          null AS res_2_remote_endpoint,
+          null AS res_2_resource_url,
+          null AS res_2_last_run,
+          null AS res_2_total_num_questions,
+          null AS res_2_total_num_answers,
+          null AS res_2_total_percent_complete,
+          null AS res_2_num_required_questions,
+          null AS res_2_num_required_answers,
+          activities_1.questions['multiple_choice_00000'].correctAnswer AS res_1_multiple_choice_00000_choice,
+          activities_1.questions['multiple_choice_01000'].correctAnswer AS res_1_multiple_choice_01000_choice,
+          activities_1.questions['multiple_choice_02000'].correctAnswer AS res_1_multiple_choice_02000_choice,
+          activities_1.questions['multiple_choice_03000'].correctAnswer AS res_1_multiple_choice_03000_choice,
+          null AS res_1_multiple_choice_03000_submitted,
+          null AS res_1_open_response_11111_text,
+          null AS res_1_open_response_22222_text,
+          null AS res_1_open_response_22222_submitted,
+          null AS res_1_image_question_33333_image_url,
+          null AS res_1_image_question_33333_text,
+          null AS res_1_image_question_33333_answer,
+          null AS res_1_image_question_44444_image_url,
+          null AS res_1_image_question_44444_text,
+          null AS res_1_image_question_44444_answer,
+          null AS res_1_image_question_44444_submitted,
+          null AS res_1_managed_interactive_55555_text,
+          activities_1.questions['managed_interactive_66666'].correctAnswer AS res_1_managed_interactive_66666_choice,
+          null AS res_1_managed_interactive_77777_image_url,
+          null AS res_1_managed_interactive_77777_text,
+          null AS res_1_managed_interactive_77777_answer,
+          null AS res_1_managed_interactive_88888_json,
+          null AS res_1_managed_interactive_88888_url,
+          null AS res_1_managed_interactive_99999_json,
+          null AS res_2_managed_interactive_88888_json,
+          null AS res_2_managed_interactive_88888_url
+        FROM activities_1, activities_2
```

Comment:
I think this from will have the same problems. When activities_1 is empty then the result of this query will be empty.

Reply:
Also fixed in https://github.com/concord-consortium/report-service/pull/178/commits/036a56584b621c5049c13ea4d0e47a031c5f91bd